physics/barycentric_rotating_reference_frame_body.hpp

```diff
@@ -85,6 +100,63 @@ BarycentricRotatingReferenceFrame<InertialFrame, ThisFrame>::secondaries()
   return secondaries_;
 }
 
+template<typename InertialFrame, typename ThisFrame>
+template<int degree>
+Derivative<Position<InertialFrame>, Instant, degree>
+BarycentricRotatingReferenceFrame<InertialFrame, ThisFrame>::
+    primary_derivative(Instant const& t) const {
+  BarycentreCalculator<Derivative<Position<InertialFrame>, Instant, degree>,
+                       GravitationalParameter>
+      result;
+  for (not_null const primary : primaries_) {
+    if constexpr (degree == 0) {
+      result.Add(ephemeris_->trajectory(primary)->EvaluatePosition(t),
+                 primary->gravitational_parameter());
+    } else if constexpr (degree == 1) {
+      result.Add(ephemeris_->trajectory(primary)->EvaluateVelocity(t),
+                 primary->gravitational_parameter());
+    } else if constexpr (degree == 2) {
+      result.Add(
+          ephemeris_->ComputeGravitationalAccelerationOnMassiveBody(primary, t),
+          primary->gravitational_parameter());
+    } else {
+      static_assert(degree == 3);
+      result.Add(ephemeris_->ComputeGravitationalJerkOnMassiveBody(primary, t),
+                 primary->gravitational_parameter());
+    }
+  }
+  return result.Get();
+}
+
+template<typename InertialFrame, typename ThisFrame>
+template<int degree>
+Derivative<Position<InertialFrame>, Instant, degree>
+BarycentricRotatingReferenceFrame<InertialFrame, ThisFrame>::
+    secondary_derivative(Instant const& t) const {
+  BarycentreCalculator<Derivative<Position<InertialFrame>, Instant, degree>,
+                       GravitationalParameter>
+      result;
+  for (not_null const secondary : secondaries_) {
+    if constexpr (degree == 0) {
+      result.Add(ephemeris_->trajectory(secondary)->EvaluatePosition(t),
+                 secondary->gravitational_parameter());
+    } else if constexpr (degree == 1) {
+      result.Add(ephemeris_->trajectory(secondary)->EvaluateVelocity(t),
+                 secondary->gravitational_parameter());
+    } else if constexpr (degree == 2) {
+      result.Add(ephemeris_->ComputeGravitationalAccelerationOnMassiveBody(
+                     secondary, t),
+                 secondary->gravitational_parameter());
+    } else {
+      static_assert(degree == 3);
+      result.Add(
+          ephemeris_->ComputeGravitationalJerkOnMassiveBody(secondary, t),
+          secondary->gravitational_parameter());
+    }
+  }
+  return result.Get();
+}
```

Comment:
I don't like all this code replication.  Let's have an implementation function that takes the vector of bodies to accumulated.